Tasks/AzureResourceGroupDeployment/operations/azure-rest/azure-arm-common.ts

```diff
@@ -15,9 +15,11 @@ export class ApplicationTokenCredentials {
     private clientId: string;
     private domain: string;
     private secret: string;
+    public armUrl: string;
+    public authUrl: string;
     private token_deferred: Q.Promise<string>;
 
-    constructor(clientId: string, domain: string, secret: string) {
+    constructor(clientId: string, domain: string, secret: string, armUrl: string, authUrl: string) {
```

Comment:
authUrl is confusing since auth is mostly used for authentication. I would prefer full name authorityUrl